init

```diff
@@ -28,27 +41,57 @@ if [ $TF_INIT_EXIT_CODE -ne 0 ]; then
   exit 1
 fi
 
-if [ -d ${TF_CONFIG_ROOT}/.terraform/modules/psoxy/ ]; then
+if [ -d ".terraform/modules/psoxy/" ]; then
+  PSOXY_BASE_DIR=".terraform/modules/psoxy/"
+elif [ -d "${TF_CONFIG_ROOT}/.terraform/modules/psoxy/" ]; then
   # use checkout of repo done by Terraform
-  PSOXY_BASE_DIR=${TF_CONFIG_ROOT}/.terraform/modules/psoxy/
+  PSOXY_BASE_DIR="${TF_CONFIG_ROOT}/.terraform/modules/psoxy/"
 else
   # use checkout of repo on your local machine
   cd ../../..
   PSOXY_BASE_DIR="`pwd`/"
-  cd ${TF_CONFIG_ROOT}
+  cd "${TF_CONFIG_ROOT}"
 fi
 
-if [ ! -f terraform.tfvars ]; then
-  printf "Initializing ${BLUE}terraform.tfvars${NC} file for your configuration ...\n"
+TFVARS_FILE="${TF_CONFIG_ROOT}/terraform.tfvars"
 
-  TFVARS_FILE="${TF_CONFIG_ROOT}/terraform.tfvars"
+if [ ! -f "${TFVARS_FILE}" ]; then
+  printf "Initializing ${BLUE}terraform.tfvars${NC} file for your configuration ...\n"
 
-  cp ${TF_CONFIG_ROOT}/terraform.tfvars.example.hcl $TFVARS_FILE
+  if [ -f "${TF_CONFIG_ROOT}/terraform.tfvars.example.hcl" ]; then
+    cp "${TF_CONFIG_ROOT}/terraform.tfvars.example.hcl" "${TFVARS_FILE}"
+  else
+    touch "${TFVARS_FILE}"
+  fi
 
-  ${PSOXY_BASE_DIR}tools/init-tfvars.sh $TFVARS_FILE $PSOXY_BASE_DIR
+  ${PSOXY_BASE_DIR}tools/init-tfvars.sh $TFVARS_FILE $PSOXY_BASE_DIR $DEPLOYMENT_ENV
 else
   printf "${RED}Nothing to initialize. File terraform.tfvars already exists.${NC}\n\n"
 fi
 
+# create example build script, to support building deployment bundle (JAR) outside of Terraform
+# (useful for debugging)
+
+BUILD_DEPLOYMENT_BUNDLE_SCRIPT=${TF_CONFIG_ROOT}/build
+if [ -f $BUILD_DEPLOYMENT_BUNDLE_SCRIPT ]; then
+  rm "$BUILD_DEPLOYMENT_BUNDLE_SCRIPT"
+fi
+
+touch "$BUILD_DEPLOYMENT_BUNDLE_SCRIPT"
+echo "#!/bin/bash" >> $BUILD_DEPLOYMENT_BUNDLE_SCRIPT
+
+# pattern used to grep for provider at top-level of Terraform configuration
+TOP_LEVEL_PROVIDER_PATTERN="^├── provider\[registry.terraform.io/hashicorp"
+AWS_PROVIDER_COUNT=$(terraform providers | grep "${TOP_LEVEL_PROVIDER_PATTERN}/aws" | wc -l)
```

Comment:
I think this should be interactive too. 
What if the customer wants to deploy in gcs but store the terraform state in a s3 bucket? will infer aws?

Reply:
no, the backend is distinct from configured providers